### PR TITLE
Helidon 2.X - Add additional licensing report formats

### PR DESCRIPTION
**Pull request by kmizuta**
Currently, the licensing report is generated in a text format. Oracle documents the attributions in a different format. You can see an example here....
https://docs.oracle.com/en/cloud/paas/app-builder-cloud/abcsl/index.html#ABCSL-GUID-125BCD30-51FC-4049-A8AB-1FDED99A0486

HTML format is provided so that the documentation team can open it in a browser and use it to copy-paste specific portions.

CSV format was originally provided to do the same. This is the process they follow for other content where the documentation team was reading this into Excel and performing copy-paste activity. However, Excel has a limit of 32767 bytes for a cell and netty's attribution exceeded that. They were unable to open this in Excel because of this. This could be removed -- if desired.

JSON format is provided so that if - in the future - they decide they want to just automate the generation of the document, they have an easily parseable format. 

I have run the following commands on both helidon-maven-plugin and licensing module.
- mvn validate -Pcheckstyle
- mvn validate -Pcopyright
- mvn verify -Pspotbugs

licensing still has violations from the spotbugs check but they are preexisting violations.

### CODE REVIEW

**Comment by kmizuta** [comment]
Joe – no worries. I’ve made the updates.

BTW – the Jackson dependencies were already there in the main dependencyManagement, but they were scoped to test originally. I’ve reverted it back to test.


From: Joe DiPol ***@***.***>
Date: Thursday, June 30, 2022 at 3:10 PM
To: oracle/helidon-build-tools ***@***.***>
Cc: Ken Mizuta ***@***.***>, Author ***@***.***>
Subject: Re: [oracle/helidon-build-tools] Helidon 2.X - Add additional licensing report formats (PR #714)

@barchetta commented on this pull request.

Ken, apologies for the late review -- I was on vacation. Could you make these changes that we previously discussed:
·         Remove CSV format feature and dependency on commons-csv (at least for this particular PR)
·         Remove dependency on Jackson (possibly hand craft json output)
Thanks!

File: licensing/src/main/java/io/helidon/build/licensing/Report.java [hunk 1]
```diff
@@ -330,10 +363,169 @@ private boolean generateAttributionFile(AttributionDocument attributionDocument,
...
+        }
+        w.write("\t}\n");
+
+        w.write("}\n");
+
+        return true;
+    }
+
+    private static String jsonEscape(String str) {
+        return str
```
Comment (barchetta) [inline licensing/src/main/java/io/helidon/build/licensing/Report.java:449]: Need a null check here. I ran into an NPE while testing with some of our license data. I added this to work-around:
```
if (str == null) {
    return "";
}
```

File: licensing/src/main/java/io/helidon/build/licensing/Report.java [hunk 2]
```diff
@@ -330,10 +363,169 @@ private boolean generateAttributionFile(AttributionDocument attributionDocument,
...
+        List<AttributionDependency> deps = attributionDocument.getDependencies();
+        Set<String> licensesUsed = new HashSet<>();
+        boolean first = true;
+        for (AttributionDependency d : deps) {
+            HashSet<String> intersection = new HashSet<>(moduleList);
+            intersection.retainAll(d.getConsumers());
+            if (moduleList.isEmpty() || !intersection.isEmpty()) {
+                w.write("<tr valign=top>");
+                w.write(String.format("<td>%s</td>", d.getName()));
+                if (includeVersion)
```
Comment (barchetta) [inline licensing/src/main/java/io/helidon/build/licensing/Report.java:479]: Please run `mvn validate  -Pcheckstyle` and fix the few, minor code style errors.

**Review by barchetta — COMMENTED** [review]
Looking good and thanks for the contribution. A couple minor things.

**Comment by kmizuta** [comment]
Updated. I should have run all the validation checks again. I made the corrections you identified  -- but there were also some new spotbug checks that I fixed. Hopefully, third times the charm.

**Review by barchetta — APPROVED** [review]

**Pull request merged**